src/test/owr/OptimisticWithdrawalRecipientV2.t.sol

```diff
@@ -0,0 +1,730 @@
+    withdrawalMock.setFailNextAddRequest(true);
+    vm.expectRevert(OptimisticWithdrawalRecipientV2.InvalidWithdrawal_Failed.selector);
+    owrETH.requestWithdrawal{value: realFee}(single, new uint64[](1));
+    withdrawalMock.setFailNextAddRequest(false);
+  }
+
+  function testRequestSingleWithdrawal() public {
+    bytes[] memory pubkeys = new bytes[](1);
+    uint64[] memory amounts = new uint64[](1);
+    bytes memory pubkey = new bytes(48);
+    uint64 amount = 1234;
+    for (uint256 i = 0; i < 48; i++) {
+      pubkey[i] = bytes1(0xAB);
+    }
+    pubkeys[0] = pubkey;
+    amounts[0] = amount;
+
+    address _user = vm.addr(0x2);
+    owrETH.grantRoles(_user, owrETH.WITHDRAWAL_ROLE());
+    uint256 realFee = withdrawalMock.fakeExponential(0);
+
+    vm.deal(_user, 1 ether);
+    vm.startPrank(_user);
+    vm.expectEmit(true, true, true, true);
+    emit WithdrawalRequested(_user, pubkey, amount);
+    owrETH.requestWithdrawal{value: 100 wei}(pubkeys, amounts);
+    vm.stopPrank();
+
+    bytes memory requestData = abi.encodePacked(pubkey, amount);
+    bytes[] memory requestsMade = withdrawalMock.getRequests();
+    assertEq(requestsMade.length, 1);
+    assertEq(requestsMade[0], requestData);
+    assertEq(address(withdrawalMock).balance, realFee);
+    assertEq(_user.balance, 1 ether - realFee);
+  }
+
+  function testRequestBatchWithdrawal() public {
+    uint256 excessFee = 100 wei;
+    uint256 expectedTotalFee;
+    uint256 numRequests = 10;
+    bytes[] memory pubkeys = new bytes[](numRequests);
+    uint64[] memory amounts = new uint64[](numRequests);
+
+    for (uint8 i = 0; i < numRequests; i++) {
+      expectedTotalFee += withdrawalMock.fakeExponential(i);
+
+      bytes memory pubkey = new bytes(48);
+      for (uint8 j = 0; j < 48; j++) {
+        pubkey[i] = bytes1(i + 1);
+      }
+      pubkeys[i] = pubkey;
+      amounts[i] = uint64(1 << i);
+    }
+
+    address _user = vm.addr(0x1);
+    owrETH.grantRoles(_user, owrETH.WITHDRAWAL_ROLE());
+
+    vm.deal(_user, expectedTotalFee + excessFee);
+    vm.startPrank(_user);
+    owrETH.requestWithdrawal{value: expectedTotalFee}(pubkeys, amounts);
+    vm.stopPrank();
+
+    bytes[] memory requestsMade = withdrawalMock.getRequests();
+    assertEq(requestsMade.length, numRequests);
+    assertEq(_user.balance, excessFee);
+    assertEq(address(withdrawalMock).balance, expectedTotalFee);
+    for (uint256 i; i < numRequests; i++) {
+      bytes memory requestData = abi.encodePacked(pubkeys[i], amounts[i]);
+      assertEq(requestsMade[i], requestData);
+    }
+  }
+
+  function testReceiveETH() public {
+    address(owrETH).safeTransferETH(1 ether);
+    assertEq(address(owrETH).balance, 1 ether);
+  }
+
+  function testReceiveTransfer() public {
+    payable(address(owrETH)).transfer(1 ether);
+    assertEq(address(owrETH).balance, 1 ether);
+  }
+
+  function testEmitOnReceiveETH() public {
+    vm.expectEmit(true, true, true, true);
+    emit ReceiveETH(1 ether);
```

Comment:
This is unexpected to me. I don't see a declared fallback/receive func on this contract? is it an inherited behaviour? if not, i don't know how important it is to have, as most ether coming into this contract will be via skimming, which doesn't trigger contract logic at all